src/QueueClientExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+
+namespace Likvido.Worker.AzureStorageQueue
+{
+    public static class QueueClientExtensions
+    {
+        public static async Task AddMessageAndCreateIfNotExistsAsync(
+            this QueueClient queueClient,
+            string message,
+            TimeSpan? visibilityTimeout = null,
+            CancellationToken cancellationToken = default)
+        {
+            if (queueClient == null)
+            {
+                throw new ArgumentNullException(nameof(queueClient));
+            }
+
+            try
+            {
+                await queueClient.SendMessageAsync(message, visibilityTimeout, cancellationToken: cancellationToken);
+                return;
+            }
+            catch (RequestFailedException exception)
+            when (exception.ErrorCode == QueueErrorCode.QueueNotFound
+               && exception.Status == 404)
+            {
+                //swallow
+            }
+
+            await queueClient.CreateIfNotExistsAsync(cancellationToken: cancellationToken);
+            await queueClient.SendMessageAsync(message, cancellationToken);
```

Comment:
This is a matter of preference, but I would probably move these two lines into the `catch`, and then remove the `return` from line 28

Reply:
Initially, I've written it using your approach but then found this version in the azure-sdk and since it less nested it probably a bit easier to read but I don't have a strong opinion on this

Reply:
I think the level of nesting here is so small that it is easy to grasp. As I said, this comes down to opinion/preference, and in this case I prefer getting rid of the return statement, as I think it can be easily missed when reading the code